test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,156 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
+					Server: "http://kuma-control-plane.example.com",
+					TLSConfig: &monitoringv1.SafeTLSConfig{
+						CA: monitoringv1.SecretOrConfigMap{
+							Secret: &v1.SecretKeySelector{
+								Key:                  "ca-cert",
+								LocalObjectReference: v1.LocalObjectReference{Name: "kuma-tls-secret"},
+							},
+						},
+						Cert: monitoringv1.SecretOrConfigMap{
+							Secret: &v1.SecretKeySelector{
+								Key:                  "tls-cert",
+								LocalObjectReference: v1.LocalObjectReference{Name: "kuma-tls-secret"},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid KumaSDConfig - Proxy Settings",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server: "http://kuma-control-plane.example.com",
+					ProxyConfig: monitoringv1.ProxyConfig{
+						ProxyURL:             ptr.To("http://proxy.example.com"),
+						NoProxy:              ptr.To("127.0.0.1,localhost"),
+						ProxyFromEnvironment: ptr.To(true),
+						ProxyConnectHeader: map[string][]v1.SecretKeySelector{
+							"Authorization": {
+								{
+									LocalObjectReference: v1.LocalObjectReference{Name: "kuma-proxy-secret"},
+									Key:                  "proxy-auth-token",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid KumaSDConfig - Missing Proxy URL",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server: "http://kuma-control-plane.example.com",
+					ProxyConfig: monitoringv1.ProxyConfig{
+						NoProxy: ptr.To("127.0.0.1,localhost"),
+					},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid KumaSDConfig - Invalid RefreshInterval Format",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server:          "http://kuma-control-plane.example.com",
+					RefreshInterval: ptr.To(monitoringv1.Duration("5x")),
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid KumaSDConfig - Follow Redirects Enabled",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server:          "http://kuma-control-plane.example.com",
+					FollowRedirects: ptr.To(true),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Valid KumaSDConfig - Required Fields",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server: "http://kuma-control-plane.example.com",
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid KumaSDConfig - Invalid Server URL",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
+				{
+					Server: "invalid-url",
+				},
+			},
+		},
+		expectedError: true,
+	},
```

Comment:
I would like to see test cases where, 
1. `Server` is not specified at all
2. `Server` value is right. i.e, a string of length >1. (similar to what you have with the "required fields" test case.)

Reply:
For example,
```Go
name: "Server URL Unspecified",
scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
				{},
          }
},
expectedError: true,
```

Reply:
After adding the `Pattern` validation,
```Go
name: "Server URL with HTTPS schema",
scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
				{ 
                                 Server: "https://example.com", 
                                 },
          }
},
expectedError: false,
```

Reply:
```Golang
name: "Server URL with HTTP schema",
scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
			KumaSDConfigs: []monitoringv1alpha1.KumaSDConfig{
				{ 
                                 Server: "http://example.com", 
                                 },
          }
},
expectedError: false,
```